website/src/components/sidebar.tsx

```diff
@@ -14,32 +14,39 @@ import * as css from "./sidebar.css"
 // Renders a sidebar on the left side of the screen containing a drawer.
 export const Sidebar = () => {
   const dialog = useDialog()
+  // State variable that checks whether this component has loaded in.
+  const [loaded, setLoaded] = useState(false)
 
-  // On load, make sure drawer appears for desktop screens.
+  // On load, make sure drawer is initially visible and is non-modal for desktop screens.
   useEffect(() => {
     dialog.setVisible(true)
+    dialog.setModal(false)
   }, [])
 
+  useEffect(() => {
+    // If this sidebar's dialog has already animated, set the loaded state to true.
+    if (dialog.animating) {
+      setLoaded(true)
+    }
+  }, [dialog.visible])
+
   return (
     <>
       <Dialog
         {...dialog}
-        className={css.drawer}
+        // If the component has already loaded in, display the drawer with transitions.
+        className={loaded ? css.drawer : css.initDrawer}
```

Comment:
This `loaded` property feels kinda like a hack, does this follow how Reakit suggests writing drawer CSS?

Reply:
So I looked into how Reakit handles animations on Disclosures, and saw you can use the [data-enter] and [data-leave] selectors to style both enter and leave animations as mentioned here: https://reakit.io/docs/disclosure/

I tried using this but it gets a little weird because the drawer should open and close with a transition, but should not open with a transition on the first page load if that makes sense? So I thought those styles conflicted if I use the same element, and decided just to switch between two drawers with these 2 diff styles based on whether the page has loaded.

i realize this does feel hacky, so if you have any other suggestions to an approach pls lmk!

Reply:
@chanmioh Shouldn't the fact that "`data-enter` is added shortly after the element is shown" let the initial render of the ToC get set without applying a transition effect while using `data-enter` and `data-leave`?

Reply:
unfortunately since i have the TOC appear visible on load, i think the `data-enter` applies to it on the initial render (so the transition effect still appears)